examples/withdraw_from_derive.py

```diff
@@ -0,0 +1,221 @@
+            }
+        )
+        estimated_gas = w3_mainnet.eth.estimate_gas(tx)
+        tx["gas"] = estimated_gas
+        required_cost = estimated_gas * max_fee + amount
+        if eth_balance < required_cost:
+            raise RuntimeError(f"Insufficient funds: need {required_cost}, have {eth_balance}")
+        w3_mainnet.eth.call(tx)
+        return tx
+
+    tx = simulate_tx()
+    tx_receipt = sign_and_send_tx(w3=w3_mainnet, tx=tx, private_key=account._private_key)
+
+    if tx_receipt.status == TxStatus.SUCCESS:
+        print("Deposit successful!")
+    else:
+        raise Exception("Deposit transaction reverted.")
+
+
+def main():
+    load_dotenv()
+
+    if (private_key := os.environ.get("ETH_PRIVATE_KEY")) is None:
+        raise ValueError("`ETH_PRIVATE_KEY` not found in env.")
+    if (wallet := os.environ.get("WALLET")) is None:
+        raise ValueError("`WALLET` not found in env.")
+
+    client = DeriveClient(
+        private_key=private_key,
+        wallet=wallet,
+        env=Environment.PROD,
+    )
+
+    account = Account.from_key(private_key)
+
+    rpc_url = "https://rpc.lyra.finance"
+    w3 = Web3(Web3.HTTPProvider(rpc_url))
+    if not w3.is_connected():
+        raise ConnectionError(f"Failed to connect to RPC at {rpc_url}")
+
+    currency = Currency.weETH
+    chain_id = ChainID.BASE
+    lyra_addresses = get_prod_lyra_addresses()
+
+    token_data = lyra_addresses.chains[ChainID.LYRA][currency]
+    connector = token_data.connectors[chain_id][TARGET_SPEED]
+
+    controller = token_data.Controller
+    deposit_hook = token_data.LyraTSAShareHandlerDepositHook
+    mintable_token = token_data.MintableToken
+
+    abi = json.loads(CONTROLLER_ABI_PATH.read_text())
+    controller_contract = get_contract(w3=w3, address=controller, abi=abi)
+    assert controller_contract.functions.hook__().call() == deposit_hook
+
+    abi = json.loads(DEPOSIT_HOOK_ABI_PATH.read_text())
+    deposit_contract = get_contract(w3=w3, address=deposit_hook, abi=abi)
+    pool_id = deposit_contract.functions.connectorPoolIds(connector).call()
+    locked_amount = deposit_contract.functions.poolLockedAmounts(pool_id).call()
+
+    token_contract = get_erc20_contract(w3, mintable_token)
+    eth_amount = 0.001
+    amount = client.web3_client.to_wei(eth_amount, "ether")
+
+    # check there are enough funds in the pool so the amount can be withdrawn
+    if not locked_amount >= amount:
+        raise RuntimeError(
+            f"Insufficient funds locked in pool: has {locked_amount}, want {amount} ({(locked_amount/amount*100):.2f}%)"
+        )
+
+    # check gas token reserve
+    gas_eth_amount = 0.00005
+    gas_amount = client.web3_client.to_wei(gas_eth_amount, "ether")
+    eth_balance = w3.eth.get_balance(account.address)
+    if eth_balance < gas_amount:
+        print("Not enough native ETH to pay for gas, bridging from mainnet.")
+        bridge_mainnet_eth_to_derive(account, amount=gas_amount)
```

Comment:
this `gas_eth_amount` should be computed, of course, and can be estimated later. It will be useful when not using the `withdraw_wrapper`, though the ETH need to be made available available on the smart contract funding wallet in order to be able to withdraw from there then. I will move this function into `bridge/transaction.py`, but we don't need it as we can pay fees using the balance of collateral we hold on Derive instead via the `withdraw_wrapper`, which is much more convenient.